### PR TITLE
Feature/show selected excerpt on map#17

### LINKED ISSUE #17: Show selected excerpt on map
Show selected excerpt (from existing excerpts) on map.
Alternative: Disable map on usage of existing excerpts.

### PR DESCRIPTION
**Pull request by wasabideveloper**
Implementation of feature #17 and fix of inheritance problem in BoundingGeometry class. (Closes #17 ).

* run tests
* tested views by hand:
  * /orders/new [GET/POST]
  * /orders/{orderId}
  * /downloads/{uuid}

**Reviewed by**:
- [x] @das-g 
- [x] @njordan-hsr 

Is it ok to merge it into this release? The issue is assigned to milestone next https://github.com/geometalab/osmaxx/milestones/next.

### CODE REVIEW

**Comment by das-g** [comment]
> Is it ok to merge it into this release? The issue is assigned to milestone next https://github.com/geometalab/osmaxx/milestones/next.

Sure. As long as a change doesn't interfere with existing functionality, we don't have to keep it out of a release. Releases don't have to be Milestone-pure, IMHO.

**Comment by wasabideveloper** [comment]
So the issue will stay on next but we merge the code now?

File: osmaxx-py/osmaxx/excerptexport/templates/excerptexport/templates/new_excerpt_export.html [hunk 1]
```diff
@@ -33,22 +33,22 @@
                         <select class="form-control" name="existing_excerpt.id" id="existing_excerpt.id" size="10" required="required">
                             <optgroup label="&nbsp;{% trans 'Personal excerpts' %} ({{ user.username }})">
                                 {% for excerpt in excerpts.own_private %}
-                                    <option value="{{ excerpt.id }}">{{ excerpt.name }}</option>
+                                    {% include 'excerptexport/partials/excerpt_option.html' with excerpt=excerpt %}
```
Comment (njordan-hsr) [inline osmaxx-py/osmaxx/excerptexport/templates/excerptexport/templates/new_excerpt_export.html]: `with excerpt=excerpt` statement wouldn't explicitly be needed here, should we remove it? (Same is true below).

Alternatively we could do something like:

```
{% with excerpts=excerpts.own_private %}
    {% include 'excerptexport/partials/excerpt_options.html' %}
{% endwith %}
```

and in `excerptexport/partials/excerpt_options.html`:

```
{% for excerpt in excerpts %}
    {% include 'excerptexport/partials/excerpt_option.html' %}
{% endfor %}
```

File: osmaxx-py/osmaxx/excerptexport/static/excerptexport/scripts/map.js [hunk 2]
```diff
@@ -1,56 +1,111 @@
...
+        }.bind(this));
+
+        // enable excerpt on map on change of form mode (existing excerpt or new excerpt)
+        this.formElementPartsSwitcher.addEventListener('change', function(event) {
+            this.locationFilter.enable();
+        }.bind(this));
+    };
+
+
+    var map = map = L.map('map').setView([0, 0], 2);
```
Comment (das-g) [inline osmaxx-py/osmaxx/excerptexport/static/excerptexport/scripts/map.js]: suspicious

File: osmaxx-py/osmaxx/excerptexport/static/excerptexport/scripts/map.js [hunk 3]
```diff
@@ -1,56 +1,111 @@
...
+        // enable excerpt on map on change of form mode (existing excerpt or new excerpt)
+        this.formElementPartsSwitcher.addEventListener('change', function(event) {
+            this.locationFilter.enable();
+        }.bind(this));
+    };
+
+
+    var map = L.map('map').setView([0, 0], 2);
+        // add an OpenStreetMap tile layer
+        L.tileLayer('http://{s}.tile.osm.org/{z}/{x}/{y}.png', {
```
Comment (das-g) [inline osmaxx-py/osmaxx/excerptexport/static/excerptexport/scripts/map.js]: Why the indent w.r.t. `var map`?

File: osmaxx-py/osmaxx/excerptexport/static/excerptexport/scripts/map.js [hunk 4]
```diff
@@ -1,56 +1,111 @@
...
+         * -> a user changed the excerpt, so change the form part to 'new-excerpt'
+         */
+        this.userChangeExcerptOnMapShowNewExcerptPart = function() {
+            var locationFilterBounds = this.locationFilter.getBounds();
+            var select = this.selectElementExistingExcerpts;
+            if(!(selectElementExistingExcerpts.value != "" &&
+                    select.querySelector('option[value="'+select.value+'"]').getAttribute('data-north') == locationFilterBounds._northEast.lat &&
+                    select.querySelector('option[value="'+select.value+'"]').getAttribute('data-west') == locationFilterBounds._southWest.lng &&
+                    select.querySelector('option[value="'+select.value+'"]').getAttribute('data-east') == locationFilterBounds._northEast.lng &&
+                    select.querySelector('option[value="'+select.value+'"]').getAttribute('data-south') == locationFilterBounds._southWest.lat)) {
```
Comment (das-g) [inline osmaxx-py/osmaxx/excerptexport/static/excerptexport/scripts/map.js]: consider extracting `select.querySelector('option[value="'+select.value+'"]').` as a local variable or `select.querySelector('option[value="'+select.value+'"]').getAttribute('data-' + direction)` as a function of `direction`

File: osmaxx-py/osmaxx/excerptexport/views.py [hunk 5]
```diff
@@ -29,9 +29,9 @@
 class NewExtractionOrderView(LoginRequiredMixin, FrontendAccessRequiredMixin, View):
     def get(self, request, excerpt_form_initial_data=None):
         active_excerpts = Excerpt.objects.filter(is_active=True)
-        active_bbox_excerpts = active_excerpts.filter(bounding_geometry__bboxboundinggeometry__isnull=False)
+        active_bbox_excerpts = active_excerpts.filter(_bounding_geometry__bboxboundinggeometry__isnull=False)
```
Comment (njordan-hsr) [inline osmaxx-py/osmaxx/excerptexport/views.py]: We're using a private 'variable' to filter by? If we should use this attribute, should it really be marked as _don't touch_/internal?

[Pep 8](https://www.python.org/dev/peps/pep-0008/#method-names-and-instance-variables)
[PEP 8 Stackoverflow Explanation](http://stackoverflow.com/questions/1641219/does-python-have-private-variables-in-classes)

**Pull request merged**